include/fastdds/dds/domain/DomainParticipant.hpp

```diff
@@ -943,6 +946,46 @@ class DomainParticipant : public Entity
      */
     bool has_active_entities();
 
+    /**
+     * Enables the monitor service in this DomainParticipant.
+     *
+     * @return RETCODE_OK if the monitor service could be correctly enabled.
+     * @return RETCODE_ERROR if the monitor service could not be enabled properly.
+     *
+     * @note Not supported yet. Currently returns RETCODE_UNSUPPORTED
+     */
+    RTPS_DllAPI ReturnCode_t enable_monitor_service() const;
+
+    /**
+     * Disable the monitor service in this DomainParticipant. Does nothing if the service was not enabled before.
+     *
+     * @return RETCODE_OK if the monitor service could be correctly disabled.
+     * @return RETCODE_NOT_ENABLED if the monitor service was not previously enabled.
+     * @return RETCODE_ERROR if the service could not be properly disabled.
+     *
+     * @note Not supported yet. Currently returns RETCODE_UNSUPPORTED
+     */
+    RTPS_DllAPI ReturnCode_t disable_monitor_service() const;
+
+    /**
+     * Enables the internal monitor service in this DomainParticipant.
+     *
+     * @param wdata Pointer to the WriterProxyData object.
+     * @param rdata Pointer to the ReaderProxyData object.
+     * @param [out] reason On return will specify the reason of failed matching (if any).
+     * @param [out] incompatible_qos On return will specify all the QoS values that were incompatible (if any).
+     *
+     * @return RETCODE_OK if qos are compatible.
+     * @return RETCODE_ERROR if qos are incompatible. Check reason and incompatible_qos.
+     *
+     * @note Not supported yet. Currently returns RETCODE_UNSUPPORTED
+     */
+    RTPS_DllAPI ReturnCode_t check_compatible_qos(
```

Comment:
Seeing as the `MatchingFailureMask` is returned as reason, should this method not be called `check_successful_matching`? The mask could return some other reason different to an incompatible QoS...

Reply:
Keeping the name as it was stated in a later comment